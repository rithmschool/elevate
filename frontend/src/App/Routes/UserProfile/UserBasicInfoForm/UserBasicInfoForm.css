/**/
.borderrounded {
  margin: 10% auto;
  background-color: #f4f6f8;
}
