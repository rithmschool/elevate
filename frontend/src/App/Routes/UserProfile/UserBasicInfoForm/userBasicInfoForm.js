import React, { useState } from "react";
import { Button, Form } from "react-bootstrap";
import moment from "moment";
import "../FormStyles.css";
import "./userBasicInfoForm.scss";

/** Update user basic info */
function UserBasicInfoForm(props) {
  const [first_name, setFirstName] = useState(props.first_name);
  const [last_name, setLastName] = useState(props.last_name);
  const [email, setEmail] = useState(props.email);
  const [current_company, setCurrentCompany] = useState(props.current_company);
  const [hire_date, setHireDate] = useState(props.hire_date);
  const [isEdit, setIsEdit] = useState(false);

  return (
    <div
      className={`user-basic-info-form container border rounded shadow pt-5`}
    >
      <Form onSubmit={handleSubmit}>
        <div className="form-styles_flex-space-between">
          <h3>Basic info</h3>

          {!isEdit && (
            <i
              data-testid="turnEditOn"
              className="m-3 fas fa-edit fa-1x"
              onClick={() => setIsEdit(true)}
            ></i>
          )}
        </div>

        <Form.Group>
          <span>First name</span>

          <Form.Control
            onChange={e => setFirstName(e.target.value)}
            id="EditUser-first_name"
            data-testid="firstName"
            name="first_name"
            type="text"
            disabled={!isEdit}
            value={first_name}
          />
        </Form.Group>

        <Form.Group>
          <span>Last name</span>

          <Form.Control
            onChange={e => setLastName(e.target.value)}
            id="EditUser-last_name"
            data-testid="lastName"
            name="last_name"
            type="text"
            disabled={!isEdit}
            value={last_name}
          />
        </Form.Group>

        <Form.Group>
          <span>Email</span>

          <Form.Control
            onChange={e => setEmail(e.target.value)}
            id="EditUser-email"
            data-testid="email"
            name="email"
            type="text"
            disabled={!isEdit}
            value={email}
          />
        </Form.Group>

        <Form.Group>
          <span>Current company</span>

          <Form.Control
            onChange={e => setCurrentCompany(e.target.value)}
            id="EditUser-current_company"
            data-testid="currentCompany"
            name="current_company"
            type="text"
            disabled={!isEdit}
            value={current_company}
          />
        </Form.Group>

        <Form.Group>
          <span>Hire date</span>

          <Form.Control
            onChange={e => setHireDate(e.target.value)}
            id="EditUser-hire_date"
            data-testid="hireDate"
            name="hire_date"
            type="date"
            disabled={!isEdit}
            value={formatHireDate() || ""}
          />
        </Form.Group>

        <div className="row justify-content-center">
          {isEdit && (
            <div>
              <Button className="login-submit mr-3 ml-3" type="submit">
                Submit
              </Button>

              <h6
                className="mr-3 ml-3 form-styles_cancel"
                data-testid="cancel"
                onClick={() => setIsEdit(false)}
              >
                Cancel
              </h6>
            </div>
          )}
        </div>
      </Form>
    </div>
  );

  function formatHireDate() {
    if (hire_date === null) {
      return null;
    } else {
      return moment(hire_date).format("YYYY-MM-DD");
    }
  }

  function handleSubmit(evt) {
    evt.preventDefault();

    const updatedUser = {
      first_name,
      last_name,
      email,
      current_company,
      hire_date: formatHireDate()
    };

    props.handleUserUpdate(updatedUser, props.userId);
    setIsEdit(false);
  }
}

export default UserBasicInfoForm;
