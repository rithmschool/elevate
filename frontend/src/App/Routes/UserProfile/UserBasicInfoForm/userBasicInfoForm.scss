.user-basic-info-form {
  .border.rounded {
    margin: 10% auto;
    background-color: #f4f6f8;
  }
}
