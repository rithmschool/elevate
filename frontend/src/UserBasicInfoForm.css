.user-basic-info-form_cancel {
  padding: 0.8em 0;
  color: #555;
  cursor: pointer;
}

.user-basic-info-form_flex-space-between {
  display: flex;
  justify-content: space-between;
}
