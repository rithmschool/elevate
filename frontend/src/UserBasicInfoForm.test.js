import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import UserBasicInfoForm from "./userBasicInfoForm";

it("renders without crashing", function() {
  shallow(<UserBasicInfoForm />);
});

it("matches snapshot", function() {
  let currentUser = {
    current_company: "Rithm",
    email: "test@gmail.com",
    first_name: "user",
    hire_date: "2019-08-13",
    last_name: "test"
  };
  let wrapper = shallow(<UserBasicInfoForm {...currentUser} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

describe("UserSalaryInfoForm", function() {
  let wrapper;
  let currentUser = {
    current_company: "Rithm",
    email: "test@gmail.com",
    first_name: "user",
    hire_date: "2019-08-13",
    last_name: "test"
  };
  beforeEach(() => {
    wrapper = mount(<UserBasicInfoForm {...currentUser} />);
  });

  it("has states", function() {
    expect(wrapper.state("first_name")).toEqual("user");
    expect(wrapper.state("last_name")).toEqual("test");
    expect(wrapper.state("email")).toEqual("test@gmail.com");
    expect(wrapper.state("current_company")).toEqual("Rithm");
    expect(wrapper.state("hire_date")).toEqual("2019-08-13");
    expect(wrapper.state("isEdit")).toEqual(false);
  });

  it("changes isEdit state to true whem click on edit", function() {
    wrapper.find("i.fa-edit").simulate("click");
    expect(wrapper.state("isEdit")).toEqual(true);
  });
});
