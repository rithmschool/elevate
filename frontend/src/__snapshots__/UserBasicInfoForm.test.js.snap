// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="EditPUserForm container border rounded"
  style={
    Object {
      "backgroundColor": "#F4F6F8",
    }
  }
>
  <Row
    tag="div"
  >
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <p>
        Basic info
      </p>
    </Col>
    <Col
      align="right"
      md={6}
      sm={
        Object {
          "offset": 0,
          "size": 8,
        }
      }
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <i
        className="fas fa-edit text-info"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "1.1em",
            "marginTop": "1em",
          }
        }
      />
    </Col>
  </Row>
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-first_name"
            name="first_name"
            onChange={[Function]}
            type="text"
            value="user"
          />
          <span>
            First name
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-last_name"
            name="last_name"
            onChange={[Function]}
            type="text"
            value="test"
          />
          <span>
            Last name
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-email"
            name="email"
            onChange={[Function]}
            type="email"
            value="test@gmail.com"
          />
          <span>
            Email
          </span>
        </Label>
      </Col>
    </Row>
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-current_company"
            name="current_company"
            onChange={[Function]}
            type="text"
            value="Rithm"
          />
          <span>
            Current company
          </span>
        </Label>
      </Col>
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-hire_date"
            name="hire_date"
            onChange={[Function]}
            type="date"
            value="2019-08-13"
          />
          <span>
            Hire date
          </span>
        </Label>
      </Col>
      <Col
        align="left"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Button
          color="info"
          disabled={true}
          size="sm"
          tag="button"
        >
          Save
        </Button>
      </Col>
    </Row>
  </Form>
</div>
`;
